### ISSUE TITLE
StaticWebAssetsFile provider doesn't support windows style paths

### ISSUE THREAD

**Issue by javiercn** [aspnet/AspNetCore#15314]
Looking at another issue (https://github.com/aspnet/AspNetCore/issues/15278) I've realized that we don't support windows style paths in the static web assets file provider. I'm leaving the comment here so I can attach it to a new issue, and triage it separately.

_Originally posted by @javiercn in https://github.com/aspnet/AspNetCore/pull/14803/review_comment/create_

**Comment by javiercn** [comment]
I think this is worth addressing because querying the WebRootFileProvider with a windows style path will cause it to fail when using static web assets. 

I believe to support this we need to normalize the path further and replace `\` with `/` at the very least, but it also presents the issue that the static web assets don't behave like the the physical file provider. 

For example, different results will be produced with the following snippet depending on whether its the physical file provider or the static web assets file provider:
```csharp
var rootContents = provider.GetDirectoryContents("");
var files = new List<IFileInfo>(rootcontents.Where(e => !e.IsDirectory));
var directory stack = new Stack<IFileInfo>(rootContents
    .Where(e => e.IsDirectory).Select(d => provider.GetFileInfo(d.Name).Cast<DirectoryInfoBase>());

while(stack.TryPop(out var currentFolder))
{
    var contents = currentFolder.EnumerateFileSystemInfos();
    files.AddRange(contents.Where(e => !e.IsDirectory));
    stack.Push(contents.Where(e => e.IsDirectory));
}
```

**Comment by ryanbrandenburg** [comment]
Fixed by #14803.
